tools/docs/tests/test_rst_check.py

```diff
@@ -13,6 +13,29 @@ def test_rst_check_current_version_constructor():
     assert version_file.path == "PATH"
 
 
+@pytest.mark.parametrize(
+    "constant",
+    (("backticks_re", "BAD_TICKS_REGEX"),
+     ("invalid_reflink_re", "INVALID_REFLINK"),
+     ("new_line_re", "VERSION_HISTORY_NEW_LINE_REGEX"),
+     ("punctuation_re", "REF_WITH_PUNCTUATION_REGEX"),
+     ("section_name_re", "VERSION_HISTORY_SECTION_NAME")))
+def test_rst_check_current_version_regexes(patches, constant):
```

Comment:
either in this Pr or a follow-up can you comment up these tests a bit?  when I broke one the error messages were pretty confusing and the code was pretty hard to read for someone with low python skills.

Reply:
yep happy to add some comments in a follow up - mostly i havent because the tests fairly systematically test methods/functions and follow pytest/mock conventions

testing can be done with, eg:

```

bazel run //tools/docs:pytest_rst_check

```

if something breaks `breakpoint()` can be a lifesaver with the tests

Reply:
...well that is not entirely accurate - i added the `patches` fixture - i should probs document how the tests work, and how to quickly debug them better more generally i think

Reply:
actually there is this here https://github.com/envoyproxy/envoy/blob/main/tools/README.md - could probs do with some updating

Reply:
https://github.com/envoyproxy/envoy/issues/17606